sdk/iot/core/README.md

```diff
@@ -12,11 +12,54 @@ TBD
 
 ## Examples
 
+### IoT Hub Client
+
+To use IoT Hub connectivity, the first action by a developer should be to initialize the
+client with the `az_iot_hub_client_init()` API. Once that is initialized, you may use the
+`az_iot_hub_client_user_name_get()` and `az_iot_hub_client_client_id_get()` to get the
+user name and client id to establish a connection with IoT Hub. An example use case is below.
+
+```C
+//FOR SIMPLICITY THIS DOES NOT HAVE ERROR CHECKING. IN PRODUCTION ENSURE PROPER ERROR CHECKING.
+
+az_iot_hub_client my_client;
+static az_span my_iothub_hostname = AZ_SPAN_LITERAL_FROM_STR("constoso.azure-devices.net");
+static az_span my_device_id = AZ_SPAN_LITERAL_FROM_STR("contoso_device");
+
+//Make sure to size the buffer to fit the user name (100 is an example)
+static uint8_t my_mqtt_user_name_buffer[100];
+static az_span my_mqtt_user_name = AZ_SPAN_LITERAL_FROM_BUFFER(my_mqtt_user_name_buffer);
+
+//Make sure to size the buffer to fit the client id (16 is an example)
+static uint8_t my_mqtt_client_id_buffer[16];

```

Comment:
Assume user has no knowledge about how `user_name` and `client_id` is built, shall we tell users how to find out the proper size for these buffers in comments?

Reply:
Yes I was thinking exactly that. I added it above the functions in the `.c` and the header file for doxygen. I can add it here too though.

Reply:
added here in the docs too